pkg/netutil/netutil.go

```diff
@@ -174,29 +174,29 @@ func URLStringsEqual(ctx context.Context, lg *zap.Logger, a []string, b []string
 	if len(a) != len(b) {
 		return false, fmt.Errorf("len(%q) != len(%q)", a, b)
 	}
-	urlsA := make([]url.URL, 0)
-	for _, str := range a {
-		u, err := url.Parse(str)
-		if err != nil {
-			return false, fmt.Errorf("failed to parse %q", str)
-		}
-		urlsA = append(urlsA, *u)
+	urlsA, err := stringsToURLs(a)
+	if err != nil {
+		return false, err
 	}
-	urlsB := make([]url.URL, 0)
-	for _, str := range b {
-		u, err := url.Parse(str)
-		if err != nil {
-			return false, fmt.Errorf("failed to parse %q", str)
-		}
-		urlsB = append(urlsB, *u)
+	urlsB, err := stringsToURLs(b)
+	if err != nil {
+		return false, err
 	}
 	if lg == nil {
 		lg, _ = zap.NewProduction()
 		if lg == nil {
 			lg = zap.NewExample()
 		}
 	}
-	return urlsEqual(ctx, lg, urlsA, urlsB)
+	sort.Sort(types.URLs(urlsA))
+	sort.Sort(types.URLs(urlsB))
+	for i := range urlsA {
+		if !reflect.DeepEqual(urlsA[i], urlsB[i]) {
+			// If urls are not equal, try to resolve it and compare again.
+			return urlsEqual(ctx, lg, urlsA, urlsB)
```

Comment:
Is there a reason to not make the 'raw' comparison a part of the 'urlsEqual' logic (just before the address resolution) ?

Reply:
Are you asking about replacing reflect with comparing the fields of the URL structure?
Or about whether there will be problems after the merge of the current pool request?

Regarding the rejection of the reflect, this can be done, there is no problem here, it just seems to me that it is more concise with reflect.

Regarding the appearance of problems after the merge, I think that there should be no problems, because the behavior changes only in one direction, there will be more valid configurations (those that turned out to be not valid now, but they are expected to be valid)